superqt/_eliding_label.py

```diff
@@ -0,0 +1,105 @@
+from typing import List
+
+from superqt.qtcompat.QtCore import QPoint, QRect, QSize, Qt
+from superqt.qtcompat.QtGui import QFont, QFontMetrics, QResizeEvent, QTextLayout
+from superqt.qtcompat.QtWidgets import QLabel
+
+
+class QElidingLabel(QLabel):
+    """A QLabel variant that will elide text (add '…') to fit width.
+
+    QElidingLabel(parent: Optional[QWidget] = None, f: Qt.WindowFlags = ...)
+    QElidingLabel(text: str, parent: Optional[QWidget] = None, f: Qt.WindowFlags = ...)
+
+    For a multiline eliding label, use `setWordWrap(True)`.  In this case, text
+    will wrap to fit the width, and only the last line will be elided.
+    When `wordWrap()` is True, `sizeHint()` will return the size required to fit the
+    full text.
+    """
+
+    ELIDE_STRING = "…"
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self._elide_mode = Qt.TextElideMode.ElideRight
+        self._text = ""
+        if args and isinstance(args[0], str):
+            self._text = args[0]
+
+    # New Public methods
+
+    def elideMode(self) -> Qt.TextElideMode:
+        """The current Qt.TextElideMode."""
+        return self._elide_mode
+
+    def setElideMode(self, mode: Qt.TextElideMode):
+        """Set the elide mode to a Qt.TextElideMode."""
+        self._elide_mode = Qt.TextElideMode(mode)  # type: ignore
+        super().setText(self._elidedText())
+
+    def fullText(self) -> str:
+        """Return the full un-elided text."""
+        return self._text
+
+    @staticmethod
+    def wrapText(text, width, font=QFont()) -> List[str]:
+        """Returns `text`, split as it would be wrapped for `width`, given `font`.
+
+        Static method.
+        """
+        tl = QTextLayout(text, font)
+        tl.beginLayout()
+        lines = []
+        while True:
+            ln = tl.createLine()
+            if not ln.isValid():
+                break
+            ln.setLineWidth(width)
+            start = ln.textStart()
+            lines.append(text[start : start + ln.textLength()])
+        tl.endLayout()
+        return lines
+
+    # Reimplemented QT methods
+
+    def setText(self, txt):
+        self._text = txt
+        super().setText(self._elidedText())
+
+    def resizeEvent(self, rEvent: QResizeEvent):
+        super().setText(self._elidedText(rEvent.size().width()))
+        rEvent.accept()
+
+    def setWordWrap(self, wrap: bool) -> None:
+        super().setWordWrap(wrap)
+        super().setText(self._elidedText())
+
+    def sizeHint(self):
```

Comment:
Docstring?

Reply:
in looking into adding these, it turns out that even the `super()` implementation didn't have a docstring, and I don't want to start creating custom docstrings for builtin methods.  currently it gives the same result as the native method which works for me

```py
In [10]: QLabel.sizeHint?
Signature: QLabel.sizeHint(self) -> PySide2.QtCore.QSize
Docstring: sizeHint(self) -> PySide2.QtCore.QSize
Type:      method_descriptor

In [11]: QElidingLabel.sizeHint?
Signature: QElidingLabel.sizeHint(self) -> PySide2.QtCore.QSize
Docstring: sizeHint(self) -> PySide2.QtCore.QSize
```

Reply:
👍🏼